File: module-git/src/main/java/com/timecat/module/git/sgit/activities/delegate/actions/CherryPickAction.java
package com.timecat.module.git.sgit.activities.delegate.actions;

import com.timecat.module.git.R;
import com.timecat.module.git.android.activities.SheimiFragmentActivity.OnEditTextDialogClicked;
import com.timecat.module.git.sgit.activities.RepoDetailActivity;
import com.timecat.module.git.sgit.database.models.Repo;
import com.timecat.module.git.sgit.repo.tasks.SheimiAsyncTask.AsyncTaskPostCallback;
import com.timecat.module.git.sgit.repo.tasks.repo.CherryPickTask;

public class CherryPickAction extends RepoAction {

    public CherryPickAction(Repo repo, RepoDetailActivity activity) {
        super(repo, activity);
    }

    @Override
    public void execute() {
        mActivity.showEditTextDialog(R.string.git_dialog_cherrypick_title,
                R.string.git_dialog_cherrypick_msg_hint,
                R.string.git_dialog_label_cherrypick,
                new OnEditTextDialogClicked() {
                    @Override
                    public void onClicked(String text) {
                        cherrypick(text);
                    }
                });
        mActivity.closeOperationDrawer();
    }

    public void cherrypick(String commit) {
        CherryPickTask task = new CherryPickTask(mRepo, commit, new AsyncTaskPostCallback() {
            @Override
            public void onPostExecute(Boolean isSuccess) {
                mActivity.reset();
            }
        });
        task.executeTask();
    }

}

File: module-git/src/main/java/com/timecat/module/git/sgit/activities/delegate/actions/CherryPickAction.kt
package com.timecat.module.git.sgit.activities.delegate.actions

import com.timecat.module.git.R
import com.timecat.module.git.sgit.activities.SheimiFragmentActivity.OnEditTextDialogClicked
import com.timecat.module.git.sgit.activities.RepoDetailActivity
import com.timecat.module.git.sgit.database.models.Repo
import com.timecat.module.git.tasks.SheimiAsyncTask.AsyncTaskPostCallback
import com.timecat.module.git.tasks.CherryPickTask

class CherryPickAction(repo: Repo, activity: RepoDetailActivity) : RepoAction(repo, activity) {
    override fun execute() {
        mActivity.showEditTextDialog(R.string.git_dialog_cherrypick_title,
            R.string.git_dialog_cherrypick_msg_hint,
            R.string.git_dialog_label_cherrypick,
            object : OnEditTextDialogClicked {
                override fun onClicked(text: String?) {
                    cherrypick(text)
                }
            })
    }

    fun cherrypick(commit: String?) {
        val task = CherryPickTask(mRepo, commit!!, object : AsyncTaskPostCallback {
            override fun onPostExecute(isSuccess: Boolean?) {
                mActivity.reset()
            }
        })
        task.executeTask()
    }
}